pkg/ccl/importccl/read_import_pgdump.go

```diff
@@ -578,11 +584,27 @@ func (m *pgDumpReader) readFile(
 				if count <= resumePos {
 					continue
 				}
-				if expected, got := len(conv.VisibleCols), len(tuple); expected != got {
+				if expected, got := expectedColLen, len(tuple); expected != got {
 					return errors.Errorf("expected %d values, got %d: %v", expected, got, tuple)
 				}
-				for i, expr := range tuple {
-					typed, err := expr.TypeCheck(ctx, &semaCtx, conv.VisibleColTypes[i])
+				conv.IsTargetCol = make(map[int]struct{})
+				colLookup := make(map[string]int, 0)
+				for j, col := range conv.VisibleCols {
+					colLookup[col.Name] = j
+				}
+				for j, expr := range tuple {
+					var colName string
+					if len(i.Columns) == 0 {
+						colName = conv.VisibleCols[j].Name
+					} else {
+						colName = i.Columns[j].String()
+					}
+					ind, ok := colLookup[colName]
```

Comment:
It is unfortunate that now the no column case for inserts is forced to go through a string map lookup. The if block above's purpose is to set colName so that the lookup can be used, but we really only need ind to be set correctly. So this should also be fine?:

```
if len(i.Columns) == 0 {
  ind = j
} else {
  colName := i.Columns[j].String();
  ind, ok = colLookup[colName];
  if !ok { ... }
}
```

Reply:
Yes you're right, I did the change as you suggested.